torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -173,14 +191,38 @@ void DistAutogradContainer::eraseContextIdAndReset(int64_t context_id) {
   }
 }
 
+void DistAutogradContainer::cleanupContextWatchdog() {
+  while (!terminateWatchdog_.load()) {
+    {
+      std::lock_guard<std::mutex> guard(autograd_context_lock_);
+      for (auto it = autograd_context_.begin(); it != autograd_context_.end();) {
+        auto now = std::chrono::high_resolution_clock::now();
+        auto diff = std::chrono::duration_cast<std::chrono::seconds>(now - std::get<1>(it->second));
+        if (diff >= cleanupContextTimeout) {
+          auto context_id = it->first;
+          it = autograd_context_.erase(it);
+          if (current_context_id_ == context_id) {
+            current_context_id_ = kInvalidContextId;
+          }
```

Comment:
We should probably clean up context ids that don't belong to this worker (only that belong to other workers). We can create something like DistAutogradContextMetadata, which holds the context, the timeout for the context and whether it was created on this node.

Also, note that `current_context_id_` here will always be `kInvalidContextId` since it is a thread local and we are in the watchdog thread.